thentos-core/src/Thentos/Backend/Core.hs

```diff
@@ -114,52 +117,58 @@ actionErrorToServantErr e = do
 
 
 class ThentosErrorToServantErr db where
-    thentosErrorToServantErr :: ThentosError db -> (Maybe (Priority, String), ServantErr)
+    -- | Convert a 'ThentosError' into a 'ServantErr'. If the first argument is 'Just' provided,
+    -- it must be invoked to create the returned error, otherwise a suitable default implementation
+    -- must be used for that purpose. This allows derived db's to adapt the details of the error
+    -- actually returned.
+    thentosErrorToServantErr :: Maybe MkServantErrFun -> ThentosError db
```

Comment:
Yes it's needed, how else should the A3-specific implementation ensure that *all* errors have the A3-specific format? Calling this function without that argument would result in a ServantErr whose JSON body already contains the wrong kind of JSON data structure.

Of course, the derived db could then decode that JSON data structure and repackage the contained information using it's own custom style, but that would be more than ugly.